specification/configuration.md

```diff
@@ -130,6 +136,31 @@ beyond the OpenTelemetry specification exist.
   - Distribution MUST default to `otlp`
   - Distribution MUST offer `jaeger-thrift-splunk` that defaults to `http://127.0.0.1:9080/v1/trace`
 
+### Real User Monitoring Libraries
+
+Real User Monitoring (RUM) instrumentation libraries cannot use environment variables for configuration.
+Instead, they MUST expose a `SplunkRum` object/class/namespace (depending on the language used) that allows
+setting the following properties (in a language-specific way: Java may use builders, Swift can use named
+parameters with default values, JavaScript can use objects, etc.):
+
+| Configuration property (default value) | Description                                                                                                                                                                                                 | Required |
+| -------------------------------------- | -----------                                                                                                                                                                                                 | -------- |
+| `beaconUrl` ()                         | RUM beacon URL, ex. `https://rum-ingest.<realm>.signalfx.com/v1/rum`. [1] [2]                                                                                                                               | Yes      |
+| `rumAccessToken` ()                    | RUM authentication token. [1]                                                                                                                                                                               | Yes      |
+| `applicationName` ()                   | Instrumented application name. [1]                                                                                                                                                                          | Yes      |
+| `globalAttributes` ()                  | OpenTelemetry [Attributes](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/common/common.md#attributes) that will be added to every span produced by the RUM library. | Yes      |
+| `allowInsecureBeacon` (false)          | Allows sending data to insecure endpoints not using https. [2]                                                                                                                                              | No       |
```

Comment:
Since we're codifying this as a spec now, I don't see a reason for this parameter at all in Browser RUM. This problem is already solved by browser vendors https://developer.mozilla.org/en-US/docs/Web/Security/Mixed_content . And provided that our backend rejects insecure incoming traffic (I don't mean redirecting to HTTPS, just straight rejection), this shouldn't be an issue for mobile RUMs either.

Reply:
It doesn't make sense for Android too, since it rejects all non-https connections by default. @johnbley is that option only for iOS?

Reply:
It's not just about ensuring that any traffic actually sent is https, it's about ensuring that developers very quickly know that they have mistyped something, rather than silently dropping data on the ground or allowing data to bleed out in plaintext across the internet only to be dropped by our ingest.  I would only believe it's not necessary in browser with demonstrable proof that browsers do the sane thing.  For android does it loudly and immediately complain or does it only complain later and only if you turn on debug logging?  For iOS this can be misconfigured, so, yes, it is necessary there.  It is certainly the case that we can simply not support the idea of `allowInsecureBeacon` on any given platform (particularly since some may make that impossible), but we **should** have active initialization-time checking that we are using https.

Reply:
As mentioned above, Android will crash your app if you try to use an insecure endpoint without specifically adding the additional permission to your app manifest.

Reply:
Another option is to keep it as a developer flag -- do not add to README but to CONTRIBUTING and make it clear it is unsupported and may be removed.

Reply:
Do we need to keep it in the spec if it's just a developer flag? If it's just a dev setting just for the iOS library then we can probably remove it from the spec.